commands/actions/publish.go

```diff
@@ -95,13 +96,10 @@ func buildFunc(cmd *cobra.Command, args []string) {
 	}
 	mustParseAndValidateTriggers(actions)
 
-	existsTsFiles, err := util.TsFilesExists(actions.Sources)
-	if err != nil {
-		logrus.Warn("\nError while detecting .ts files in sources")
-	}
 	tsConfigExists := util.TsConfigExists(actions.Sources)
 
-	if existsTsFiles && !tsConfigExists {
+	if anyFunctionTsFileExists(actions) && !tsConfigExists {
+		err := errors.New("Some of the functions are with .ts extension but there is no typescript config file!")
```

Comment:
I would return the concrete file that returned the error so the user doesn't have to hunt around for it.